EdgeModuleMessageSample/EdgeModuleMessageSample.cs

```diff
@@ -0,0 +1,98 @@
+
+            _moduleClient.ConnectionStatusChangeCallback = OnConnectionStatusChanged;
+            await _moduleClient.OpenAsync(cts.Token).ConfigureAwait(false);
+
+            // Now setting a callback for receiving a message from the module queue.
+            await _moduleClient.SetIncomingMessageCallbackAsync(PrintMessage).ConfigureAwait(false);
+            Console.WriteLine($"\n{DateTime.Now}> Subscribed to receive module messages over callback.");
+
+            // Now wait to receive module messages through the callback.
+            Console.WriteLine($"\n{DateTime.Now}> Module waiting to receive messages from the hub...");
+
+            // Now sending message to the module itself.
+            var message = new TelemetryMessage(Encoding.ASCII.GetBytes("sample message"));
+            await _moduleClient.SendTelemetryAsync("toNextMod", message).ConfigureAwait(false);
+            Console.WriteLine($"\n{DateTime.Now}> Sent telemetry message to the module.");
+
+            // Now continue to send messages to the module with every key press of 'M'.
+            while (!cts.IsCancellationRequested)
+            {
+                if (Console.ReadKey().Key == ConsoleKey.M)
+                {
+                    message = new TelemetryMessage(Encoding.ASCII.GetBytes("sample message"));
+                    await _moduleClient.SendTelemetryAsync(_outputTarget, message).ConfigureAwait(false);
+                }
+            }
+
+            // Now disposing module client.
+            await _moduleClient.DisposeAsync().ConfigureAwait(false);
+        }
+
+        private static void OnConnectionStatusChanged(ConnectionStatusInfo connectionStatusInfo)
+        {
+            Console.WriteLine($"\nConnection status changed to {connectionStatusInfo.Status}.");
+            Console.WriteLine($"Connection status changed reason is {connectionStatusInfo.ChangeReason}.\n");
+        }
+
+        private static Task<MessageAcknowledgement> PrintMessage(IncomingMessage receivedMessage)
+        {
+            bool messageDeserialized = receivedMessage.TryGetPayload(out string messageData);
+
+            if (messageDeserialized)
+            {
+                var formattedMessage = new StringBuilder($"Received message: [{messageData}]\n");
+
+                // User set application properties can be retrieved from the Message.Properties dictionary.
+                foreach (KeyValuePair<string, string> prop in receivedMessage.Properties)
+                {
+                    formattedMessage.AppendLine($"\tProperty: key={prop.Key}, value={prop.Value}");
+                }
+
+                Console.WriteLine($"{DateTime.Now}> {formattedMessage}");
+            }
+            else
+            {
+                Console.WriteLine($"Could not deserialize the received message. Please check your serializer settings.");
+            }
+            return Task.FromResult(MessageAcknowledgement.Complete);
```

Comment:
I think this comment may be helpful for people who aren't very familiar with async/await.

```suggestion

            // The method signature requires a Task<MessageAcknowledgement> return value, but as this sample
            // does not need to make any async calls, we simply return with Task.FromResult().
            return Task.FromResult(MessageAcknowledgement.Complete);
```